### PR TITLE
Guidelines: translations 

### PR DESCRIPTION
**Pull request by hobovsky**
I would like this to be a set of official quality guidelines which reviewers could use to determine if a Translation is of sufficient quality to be approved.

Any remarks or reviews are welcome!
Discussion: #134 
Deploy preview: [here](https://deploy-preview-153--reverent-edison-2864ea.netlify.app/recipes/authoring/translation/#general-translation-guidelines)

### CODE REVIEW

File: content/recipes/authoring/translation.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Failing to comply with the below guidelines should be reported as an issue or a
 ## General translation guidelines
 
 - **Respect [General Content Guidelines](/recipes/authoring/general/).**
-- **Translations should improve the overall quality of the kata.**. Sub-optimal quality of existing, approved translations should not be used as an argument for low quality of the translation. In particular:
+- **Translations should improve the overall quality of the kata.** Sub-optimal quality of existing, approved translations should not be used as an argument for low quality of the translation. In particular:
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:19]:
```suggestion
- **Translations should improve the overall quality of the kata.** The sub-optimal quality of existing, approved translations should not be used as an excuse for authoring a low-quality translation. In particular:
```

File: content/recipes/authoring/translation.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Failing to comply with the below guidelines should be reported as an issue or a
...
   - Description can be clarified, if necessary.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:20]:
```suggestion
- The description can be clarified if necessary.
```

File: content/recipes/authoring/translation.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ Failing to comply with the below guidelines should be reported as an issue or a
...
   - Test suite can use different structure than other translations. **[Random tests](/recipes/authoring/kata-snippets/full-tests/#random-tests), [full test suite](/recipes/authoring/kata-snippets/full-tests/) and  [sample tests](/recipes/authoring/kata-snippets/sample-tests/) should conform to Codewars quality guidelines.**
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:21]:
```suggestion
- The test suite may employ a different structure from that of other translations. **The [sample](/recipes/authoring/kata-snippets/sample-tests/) and [submit](/recipes/authoring/kata-snippets/full-tests/) tests (including [random tests](/recipes/authoring/kata-snippets/full-tests/#random-tests)) should conform to Codewars quality guidelines.**
```

File: content/recipes/authoring/translation.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ Failing to comply with the below guidelines should be reported as an issue or a
...
 - If, while translation is created, some serious problems are spotted in other language versions, they should be fixed or raised as appropriate issues in the kata discourse.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:22]:
```suggestion
- If serious issues with other language versions are uncovered in the course of authoring a translation, they should be fixed or raised as appropriate issues in the kata discourse.
```

File: content/recipes/authoring/translation.md [hunk 5]
```diff
@@ -41,7 +41,7 @@ Translation should not substantially change overall idea of the kata. It should
 
 ## Code components
 
-- **Respect [General Coding Guidelines](/recipes/authoring/kata-snippets/coding-general/), coding guidelines of your [language](/languages/), and guidelines specific for particular kata code components ([sample tests](/recipes/authoring/kata-snippets/sample-tests/), [full test suite](/recipes/authoring/kata-snippets/full-tests/), [preloaded code](/recipes/authoring/kata-snippets/preloaded/) and [proposed solution](/recipes/authoring/kata-snippets/proposed-solution/)).
+- **Respect [General Coding Guidelines](/recipes/authoring/kata-snippets/coding-general/), coding guidelines of your [language](/languages/), and guidelines specific for particular kata code components** ([sample tests](/recipes/authoring/kata-snippets/sample-tests/), [full test suite](/recipes/authoring/kata-snippets/full-tests/), [preloaded code](/recipes/authoring/kata-snippets/preloaded/) and [proposed solution](/recipes/authoring/kata-snippets/proposed-solution/)).
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:44]:
```suggestion
- **Respect the [General Coding Guidelines](/recipes/authoring/kata-snippets/coding-general/), any conventions imposed by your [language](/languages/) and guidelines specific to each component of the kata**, e.g. the [preloaded](/recipes/authoring/kata-snippets/preloaded/) section, [complete solution](/recipes/authoring/kata-snippets/proposed-solution/), [sample tests](/recipes/authoring/kata-snippets/sample-tests/) and [submit tests](/recipes/authoring/kata-snippets/full-tests/).
```

File: content/recipes/authoring/translation.md [hunk 6]
```diff
@@ -41,7 +41,7 @@ Translation should not substantially change overall idea of the kata. It should
...
 - If possible, **sample tests and fixed tests should use the same inputs in all translations.** It's not always possible between languages, but when done, it makes support of users much easier. When a question about failed test pops up in kata discourse, it can save follow-up questions for attempted language, etc.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:45]:
```suggestion
- If possible, **sample tests and fixed tests should remain consistent between translations.** It may not always be feasible due to inherent differences in programming languages, but doing so can save follow-up questions on the language a user is attempting the kata in when a question on failing tests is raised in the kata discourse.
```

File: content/recipes/authoring/translation.md [hunk 7]
```diff
@@ -16,7 +16,7 @@ Failing to comply with the below guidelines should be reported as an issue or a
 ## General translation guidelines
 
 - **Respect [General Content Guidelines](/recipes/authoring/general/).**
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:18]:
```suggestion
- **Respect the [General Content Guidelines](/recipes/authoring/general/).**
```

**Review by DonaldKellett — CHANGES_REQUESTED** [review]
Apart from the usual grammatical corrections and wording enhancements, may we rename the following terms?

- "Full test suite" --> "Submit tests"
- "Proposed solution" --> "Complete solution" / "Reference solution"

File: content/recipes/authoring/translation.md (hunk 6 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:45]: English is difficult.

**Comment by hobovsky** [comment]
Let's discuss your question here: https://github.com/codewars/docs/issues/154

File: content/recipes/authoring/translation.md [hunk 8]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+
+The guidelines should be used by translators and reviewers to verify whether translation which is about to be introduced into Codewars is of sufficient quality. Conformance with them should be verified before the translation is published, and before it's approved.
+
+Failing to comply with the below guidelines should be reported as an issue or a comment requesting for a fix. In case of severe violations, affected translation can be rejected, or even deleted.
+
+## General translation guidelines
+
+- **Respect the [General Content Guidelines](/recipes/authoring/general/).**
+- **Translations should improve the overall quality of the kata.** The sub-optimal quality of existing, approved translations should not be used as an excuse for authoring a low-quality translation. In particular:
+- The description can be clarified if necessary.
```
Comment (kazk) [inline content/recipes/authoring/translation.md:20]:
```suggestion
  - The description can be clarified if necessary.
```

File: content/recipes/authoring/translation.md [hunk 9]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- If possible, your translation should hold to **original author's idea**. In case of doubts, you can always ask them for opinion, or use initial language version as a baseline. However, especially for older kata, it may happen that initial version (and other older versions) are outdated, and concept of the kata evolved. You can ask kata contributors or other users for opinion and ideas how to proceed.
+- **Avoid translations to languages which cannot support the idea of the kata.** For example, there's no use in translating a big integer arithmetic kata to Python, which supports arbitrary precision integers natively. Do not translate kata into languages where difficulty would signifficantly differ: from high level languages to NASM or BF, or into lnaguages where kata can be solved with easy to use, advanced features of standard library (or other available libraries).
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:31]:
```suggestion
- **Avoid translations to languages which cannot support the idea of the kata.** For example, there's no use in translating a big integer arithmetic kata to Python, which supports arbitrary precision integers natively. Do not translate kata into languages where the difficulty would significantly differ: from high level languages to NASM or BF, or into languages where the kata can be solved with easy to use, advanced features of the standard library (or other available libraries).
```

File: content/recipes/authoring/translation.md [hunk 10]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+
+
+## Description
+
+- **Respect guidelines for [Writing a Kata Description](/recipes/authoring/kata-snippets/description/).**
+- **Use [Codewars Markdown Extensions](/references/markdown/extensions/#sequential-code-blocks)** to organize language specific parts of description, if present. Add sections specific to your language if necessary.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:37]:
```suggestion
- **Use [Codewars Markdown Extensions](/references/markdown/extensions/#sequential-code-blocks)** to organize language specific parts of the description, if present. Add sections specific to your language if necessary.
```

File: content/recipes/authoring/translation.md [hunk 11]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+## Code components
+
+- **Respect [General Coding Guidelines](/recipes/authoring/kata-snippets/coding-general/), coding guidelines of your [language](/languages/), and guidelines specific for particular kata code components** ([sample tests](/recipes/authoring/kata-snippets/sample-tests/), [full test suite](/recipes/authoring/kata-snippets/full-tests/), [preloaded code](/recipes/authoring/kata-snippets/preloaded/) and [proposed solution](/recipes/authoring/kata-snippets/proposed-solution/)).
+- If possible, **sample tests and fixed tests should remain consistent between translations.** It may not always be feasible due to inherent differences in programming languages, but doing so can save follow-up questions on the language a user is attempting the kata in when a question on failing tests is raised in the kata discourse.
+- Sometimes, description is not the best place to put some language specific information. For example, translator wants to add some information on arguments, data formats, memory management, etc. and keep the description language-agnostic at the same time. In such cases, it's allowed to put such information as an explanatory comment in "solution setup" snippet.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:46]:
```suggestion
- Sometimes, the description is not the best place to put some language specific information. For example, the translator may wish to add some information on arguments, data formats, memory management, etc. and keep the description language-agnostic at the same time. In such cases, it's allowed to put such information as an explanatory comments in the solution setup.
```

File: content/recipes/authoring/translation.md [hunk 12]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- If possible, **sample tests and fixed tests should remain consistent between translations.** It may not always be feasible due to inherent differences in programming languages, but doing so can save follow-up questions on the language a user is attempting the kata in when a question on failing tests is raised in the kata discourse.
+- Sometimes, description is not the best place to put some language specific information. For example, translator wants to add some information on arguments, data formats, memory management, etc. and keep the description language-agnostic at the same time. In such cases, it's allowed to put such information as an explanatory comment in "solution setup" snippet. 
+
+
+## Review process
+
+- **Translation must be reviewed before it's approved.** If reviewer is not familiar with the language of the translation, they should ask other users for help.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:51]:
```suggestion
- **The translation must be reviewed before it's approved.** If the reviewer is not familiar with the language of the translation, they should ask other users for help.
```

File: content/recipes/authoring/translation.md [hunk 13]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- **Any issues found during review should be posted** as translation comments.
+- Unfortunately, approval of translation cannot be easily blocked by raising issues. If translation has many issues, cannot be easily fixed, and there's justified concern that it could be approved prematurely while not keeping up to quality standards, **it should be rejected**.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:53]:
```suggestion
- Unfortunately, the approval of a translation cannot be easily blocked by raising issues. If a translation has many issues, cannot be easily fixed, and there's justified concern that it could be approved prematurely while not keeping up to quality standards, **it should be rejected**.
```

File: content/recipes/authoring/translation.md [hunk 14]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- After translation is approved, it becomes **responsibility of all the parties: the approver, the translator, and the kata author** (in this order) to fix any potential issues which might come up later, when users attempt to submit their solutions.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:54]:
```suggestion
- After a translation is approved, it becomes **the responsibility of all parties involved: the user who approved the translation, the translator, and the kata author** (in this order) to fix any potential issues which might come up later when users attempt to submit their solutions.
```

**Review by DonaldKellett — CHANGES_REQUESTED** [review]

File: content/recipes/authoring/translation.md [hunk 15]
```diff
@@ -9,7 +9,7 @@ next: /recipes/authoring/kumite/
 
 This article contains a set of guidelines, which can be used by translators to create good translations to existing kata. They were collected to help ensure that translations are of sufficient quality and users' experience will be as good as possible.
 
-The guidelines should be used by translators and reviewers to verify whether translation which is about to be introduced into Codewars is of sufficient quality. Conformance with them should be verified before the translation is published, and before it's approved.
+The guidelines should be used by translators and reviewers to verify whether translations about to be introduced into Codewars are of sufficient quality. Conformance with them should be verified before the translation is published and before it's approved.
 
 Failing to comply with the below guidelines should be reported as an issue or a comment requesting for a fix. In case of severe violations, affected translation can be rejected, or even deleted.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:14]:
```suggestion
Failure to comply with the below guidelines is considered an issue and should be reported as such. In case of severe violations, the affected translation can be rejected or even deleted.
```

File: content/recipes/authoring/translation.md [hunk 16]
```diff
@@ -24,17 +24,17 @@ Failing to comply with the below guidelines should be reported as an issue or a
 
 ## Conformance to overall kata idea
 
-Translation should not substantially change overall idea of the kata. It should be as consistent as possible with the original version in terms of difficulty, required performance, and overall composition. In particular:
+A translation should not substantially change the overall idea of the kata. It should be as consistent as possible with the original version in terms of difficulty, required performance, and overall composition. In particular:
 
 - **Respect the performance requirements of the kata.** If, for example, it's agreed that kata should accept only (sub)linear solutions, make sure that translated tests reject solutions of higher computational complexity. Languages and runtimes differ in terms of performance, so increase/decrease amount of test cases and input ranges when necessary.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:29]:
```suggestion
- **Respect the performance requirements of the kata.** If, for example, it's agreed that the kata should accept only (sub)linear solutions, make sure that the translated tests reject solutions of higher computational complexity. Languages and runtimes differ in terms of performance, so increase/decrease the amount of test cases and input ranges accordingly.
```

**Review by DonaldKellett — CHANGES_REQUESTED** [review]

**Review by DonaldKellett — APPROVED** [review]

File: content/recipes/authoring/translation.md [hunk 17]
```diff
@@ -41,7 +41,7 @@ A translation should not substantially change the overall idea of the kata. It s
...
 - If possible, **sample tests and fixed tests should remain consistent between translations.** It may not always be feasible due to inherent differences in programming languages, but doing so can save follow-up questions on the language a user is attempting the kata in when a question on failing tests is raised in the kata discourse.
 - Sometimes, the description is not the best place to put some language specific information. For example, the translator may wish to add some information on arguments, data formats, memory management, etc. and keep the description language-agnostic at the same time. In such cases, it's allowed to put such information as an explanatory comments in the solution setup.
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:46]:
```suggestion
- Sometimes, the description is not the best place to put some language specific information. For example, the translator may wish to add some information on arguments, data formats, memory management, etc. and keep the description language-agnostic at the same time. In such cases, it's allowed to put such information as an explanatory comment in the solution setup.
```

**Review by DonaldKellett — CHANGES_REQUESTED** [review]

**Review by DonaldKellett — APPROVED** [review]

**Review by DonaldKellett — APPROVED** [review]

**Review by DonaldKellett — APPROVED** [review]

**Comment by hobovsky** [comment]
Just like #162 , not much going on here. Good to go?

File: content/recipes/authoring/translation.md [hunk 18]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+This article contains a set of guidelines, which can be used by translators to create good translations to existing kata. They were collected to help ensure that translations are of sufficient quality and users' experience will be as good as possible.
+
+The guidelines should be used by translators and reviewers to verify whether translations about to be introduced into Codewars are of sufficient quality. Conformance with them should be verified before the translation is published and before it's approved.
+
+Failure to comply with the below guidelines is considered an issue and should be reported as such. In case of severe violations, the affected translation can be rejected or even deleted.
+
+## General translation guidelines
+
+- **Respect the [General Content Guidelines](/recipes/authoring/general/).**
+- **Translations should improve the overall quality of the kata.** The sub-optimal quality of existing, approved translations should not be used as an excuse for authoring a low-quality translation. In particular:
```
Comment (ggorlen) [inline content/recipes/authoring/translation.md:19]: Great point!

File: content/recipes/authoring/translation.md [hunk 19]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+  - The description can be clarified if necessary.
+  - The test suite may employ a different structure from that of other translations. **The [sample](/recipes/authoring/kata-snippets/sample-tests/) and [submission](/recipes/authoring/kata-snippets/full-tests/) tests (including [random tests](/recipes/authoring/kata-snippets/full-tests/#random-tests)) should conform to Codewars quality guidelines.**
+  - If serious issues with other language versions are uncovered in the course of authoring a translation, they should be fixed or raised as appropriate issues in the kata discourse.
+
+
+## Conformance to overall kata idea
```
Comment (ggorlen) [inline content/recipes/authoring/translation.md:25]: Perhaps "conformity" is more typical than "conformance"?

**Review by ggorlen — APPROVED** [review]

**Comment by Blind4Basics** [comment]
I don't like cross reviewing, so I'll do it after the last suggestions have been examined... x)

(same for the preloaded PR)

File: content/recipes/authoring/translation.md (hunk 19 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:25]:
```suggestion
## Conformity to overall kata idea
```

Well, it's not me who speaks English here, so I believe you know better :)

File: content/recipes/authoring/translation.md (hunk 18 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:19]: Thx

**Review by ggorlen — APPROVED** [review]

**Review by DonaldKellett — APPROVED** [review]

File: content/recipes/authoring/translation.md [hunk 20]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- **Avoid translations to languages which cannot support the idea of the kata.** For example, there's no use in translating a big integer arithmetic kata to Python, which supports arbitrary precision integers natively. Do not translate kata into languages where the difficulty would significantly differ: from high level languages to NASM or BF, or into languages where the kata becomes much easier because it can be solved with easy to use, advanced features of the standard library (or other available libraries).
+
+
+## Description
+
+- **Respect guidelines for [Writing a Kata Description](/recipes/authoring/kata-snippets/description/).**
+- **Use [Codewars Markdown Extensions](/references/markdown/extensions/#sequential-code-blocks)** to organize language specific parts of the description, if present. Add sections specific to your language if necessary.
+- If possible, **avoid displaying many language variants to the user**. Use conditional paragraphs to display information revelant to the currently viewed language.
```
Comment (Blind4Basics) [inline content/recipes/authoring/translation.md:38]:
```suggestion
- If possible, **avoid displaying many language variants to the user**. Use [conditional paragraphs](references/markdown/extensions/#conditional-rendering) to display information revelant to the currently viewed language.
```

File: content/recipes/authoring/translation.md [hunk 21]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- If the language version can pass tests successfuly only when run with some specific language runtime, it should be specified in the description.
```
Comment (Blind4Basics) [inline content/recipes/authoring/translation.md:39]: I don't get this sentence => ?

You mean time constraints that are different from the timeout itself?

File: content/recipes/authoring/translation.md [hunk 22]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+
+
+## Code components
+
+- **Respect the [General Coding Guidelines](/recipes/authoring/kata-snippets/coding-general/), any conventions imposed by your [language](/languages/) and guidelines specific to each component of the kata**, e.g. the [preloaded](/recipes/authoring/kata-snippets/preloaded/) section, [reference solution](/recipes/authoring/kata-snippets/reference-solution/), [sample tests](/recipes/authoring/kata-snippets/sample-tests/) and [submission tests](/recipes/authoring/kata-snippets/full-tests/).
```
Comment (Blind4Basics) [inline content/recipes/authoring/translation.md:44]:
```suggestion
- **Respect the [General Coding Guidelines](/recipes/authoring/kata-snippets/coding-general/), conventions imposed by your [language](/languages/) and guidelines specific to each component of the kata**, e.g. the [preloaded](/recipes/authoring/kata-snippets/preloaded/) section, [reference solution](/recipes/authoring/kata-snippets/reference-solution/), [sample tests](/recipes/authoring/kata-snippets/sample-tests/) and [submission tests](/recipes/authoring/kata-snippets/full-tests/).
```

Loads of links in this one. That makes it very cumbersome to read. Maybe one single link toward "Kata elements/overwiew"?

File: content/recipes/authoring/translation.md [hunk 23]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- If possible, **sample tests and fixed tests should remain consistent between translations.** It may not always be feasible due to inherent differences in programming languages, but doing so can save follow-up questions on the language a user is attempting the kata in when a question on failing tests is raised in the kata discourse.
```
Comment (Blind4Basics) [inline content/recipes/authoring/translation.md:45]: I'd reword this one the other way around:

> sample and fixed tests should be the exact same than...
> ...excpet in some rare cases where...

File: content/recipes/authoring/translation.md [hunk 24]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- Sometimes, the description is not the best place to put some language specific information. For example, the translator may wish to add some information on arguments, data formats, memory management, etc. and keep the description language-agnostic at the same time. In such cases, it's allowed to put such information as an explanatory comment in the solution setup.
+
+
+## Review process
+
+- **The translation must be reviewed before it's approved.** If the reviewer is not familiar with the language of the translation, they should ask other users for help.
```
Comment (Blind4Basics) [inline content/recipes/authoring/translation.md:51]:
```suggestion
- **Translations must be reviewed before they are approved.** If the reviewer is not familiar with the language of the translation, they should ask other users for help.
```

File: content/recipes/authoring/translation.md [hunk 25]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- **Any issues found during review should be posted** as translation comments.
```
Comment (Blind4Basics) [inline content/recipes/authoring/translation.md:52]: shouldn't that be "any issue"? (singular)

I'm not entirely sure about the meaning of "issue" in this sentence. If that's the usual "cw issue", then it should be posted in the discourse section, not in the translation. So is it rather "issues encountered by the translator so that his translation is not fully ready?"

If the latter, I guess a smaal rewording would be good to avoid confusion.

File: content/recipes/authoring/translation.md [hunk 26]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- Unfortunately, the approval of a translation cannot be easily blocked by raising issues. If a translation has many issues, cannot be easily fixed, and there's justified concern that it could be approved prematurely while not keeping up to quality standards, **it should be rejected**.
```
Comment (Blind4Basics) [inline content/recipes/authoring/translation.md:53]:
```suggestion
- If a translation has too many issues or cannot be easily fixed, and there's justified concern that it could be approved prematurely while not keeping up to quality standards, **it can be rejected**.
```

no need to emphasises the hole in the process x) => just state what can happen/be done.

note: "should" -> "can" because the current guidelines is fro the translator more than the reviewer

File: content/recipes/authoring/translation.md [hunk 27]
```diff
@@ -7,4 +7,48 @@ next: /recipes/authoring/kumite/
...
+- After a translation is approved, it becomes **the responsibility of all parties involved: the user who approved the translation, the translator, and the kata author** (in this order) to fix any potential issues which might come up later when users attempt to submit their solutions.
```
Comment (Blind4Basics) [inline content/recipes/authoring/translation.md:54]: I disagree on the order, in this one.
I understand the reason (showing the great responsability of the approver), but the first responsible should imo be the translator himself. If the pressure is on the guy who didn't write the code, that doesn't sound reasonnable to me. And if it's officially documented that way:
* some morons (translators) will hide behind that
* if the kata author trusted the translator while he doesn't know the language, he cannot actually do anything so the order is biased
* if it actually turns that way and the approver actually has to handle the mess, very few translations will be aproved in the near future...

File: content/recipes/authoring/translation.md (hunk 22 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:44]: You mean difficult to read in markdown, or on rendered page?

File: content/recipes/authoring/translation.md (hunk 26 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:53]: I hoped them to be useful for both: aurhors while creating, and reviewers for verification. But yeah, wording usually suggests author.

File: content/recipes/authoring/translation.md (hunk 27 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:54]: Sounds very reasonable, but I wanted to make approves aware that they are not just pushing some green buttons for fun.

File: content/recipes/authoring/translation.md (hunk 22 above)
Comment (Blind4Basics) [reply content/recipes/authoring/translation.md:44]: rendered

File: content/recipes/authoring/translation.md (hunk 26 above)
Comment (Blind4Basics) [reply content/recipes/authoring/translation.md:53]: yes sure. But iirc, there is a "dedicated doc to be" about translations approval. That's why I suggests this here.

File: content/recipes/authoring/translation.md (hunk 27 above)
Comment (Blind4Basics) [reply content/recipes/authoring/translation.md:54]: Yup, I got that. It's effectively hard to state this clearly.
Maybe it's just not the appropriate place? Like, keeping here focus on the translator, and in the "approving/reviewing translations" doc, make strong evidence about the responsibility of the approver/reviewer?

File: content/recipes/authoring/translation.md (hunk 21 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:39]:
```suggestion
- If a translation can pass tests successfully only when run with some specific language runtime, it should be specified in the description.
```

There used to be cases when some kata happened (was designed to?) pass, for example, with C++ 17, but not with C++ 14. Or C# Mono runtime vs C# 8. For some reason, sometimes these language versions were still available (and sometimes preselected) in the trainer, but it was not possible to pass when they were used.

File: content/recipes/authoring/translation.md (hunk 25 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:52]: It's about issues found during **review**, so on a pending translation. There's no **issue** label there :(

Comment (Blind4Basics) [reply content/recipes/authoring/translation.md:52]: oh.... x)

So maybe here again, we could talk about this more from the translator point of view: 

> if issues with the translation are found during its review, they will be discussed inside the comment section of the translation so that the translator can be notified and handle them.

or somethigng like that?

Comment (hobovsky) [reply content/recipes/authoring/translation.md:52]:
```suggestion
When issues with the translation are found during its review, they should be reported and discussed in the comment section of the translation so that the translator can be notified and handle them.
```

Feel free to edit this suggestion if you have better idea.

Comment (Blind4Basics) [reply content/recipes/authoring/translation.md:52]: perfect :+1:

File: content/recipes/authoring/translation.md (hunk 23 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:45]:
```suggestion
- **Sample tests and fixed tests should remain consistent between translations**, except in some rare cases when it's not feasible due to inherent differences in programming languages. It can save follow-up questions on the language a user is attempting the kata in when a question on failing tests is raised in the kata discourse.
```

File: content/recipes/authoring/translation.md (hunk 24 above)
Comment (Blind4Basics) [reply content/recipes/authoring/translation.md:51]: ok, I begin to get a better idea about your approch when writing this part. Removing plural, so. Forget about this one.

File: content/recipes/authoring/translation.md (hunk 26 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:53]: The doc you mean somehow shifted and drifted and split into two parts. Quality guidelines (the ones you are reading) are meant to be applicable for authors, *and* for reviewers, as a kind of checklist they can use before approving new content.

I will review wording of all existing guidelines and try to balance it so they do not suggest authors as much, and take approves more into picture.

I was also thinking about another document, kind of manual, which would describe authoring and review process from functional side: views, editors, actions, options, etc. I have no good idea for this yet.

File: content/recipes/authoring/translation.md (hunk 27 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md:54]: I think I will change the order as you suggest, and remove the strong note about the order.

Comment (hobovsky) [reply content/recipes/authoring/translation.md:54]:
```suggestion
- After a translation is approved, it becomes **the responsibility of all parties involved: the translator, the user who approved the translation, and the kata author**, to fix any potential issues which might come up later when users attempt to submit their solutions.
```

File: content/recipes/authoring/translation.md (hunk 22 above)
Comment (hobovsky) [reply content/recipes/authoring/translation.md]: This part got reorganized, let's see how it turns out.

**Review by DonaldKellett — APPROVED** [review]

File: content/recipes/authoring/translation.md [hunk 28]
```diff
@@ -19,12 +19,12 @@ Failure to comply with the below guidelines is considered an issue and should be
...
-  - If serious issues with other language versions are uncovered in the course of authoring a translation, they should be fixed or raised as appropriate issues in the kata discourse.
+  - If serious issues with other language versions are uncovered while translating, they should be fixed in the translation and appropriate issues should be raised about all other concerned languages in the kata discourse.
 
 
 ## Conformity to overall kata idea
 
-A translation should not substantially change the overall idea of the kata. It should be as consistent as possible with the original version in terms of difficulty, required performance, and overall composition. In particular:
+A translation should not substantially change either the overall idea nor the requirements of the kata. It should be as consistent as possible with the original version in terms of difficulty, required performance, and overall composition. In particular:
```
Comment (DonaldKellett) [inline content/recipes/authoring/translation.md:27]:
```suggestion
A translation should substantially change neither the overall idea nor the requirements of the kata. It should be as consistent as possible with the original version in terms of difficulty, required performance, and overall composition. In particular:
```

**Review by DonaldKellett — CHANGES_REQUESTED** [review]

**Review by DonaldKellett — APPROVED** [review]

File: content/recipes/authoring/translation.md (hunk 27 above)
Comment (Blind4Basics) [reply content/recipes/authoring/translation.md]: :+1:

**Comment by hobovsky** [comment]
It's gotten quiet here, so... merging!
